### DISCUSSION TITLE
Incorporate /mike-gimelfarb/numerical-integration algorithms for estimating limits of sequences and series.

### DISCUSSION THREAD

**Discussion by axkr** [Hipparchus-Math/hipparchus#352] [Ideas]
Can the algorithms "for estimating limits of sequences and series" in the `series` subpackage from
- https://github.com/mike-gimelfarb/numerical-integration

be incorporated into `hipparchus`?

@mike-gimelfarb as I'ver understood, you are willing to allow the series algorithm parts being published under Apache license, because it's written by yourself?
@maisonobe how must such a move be done, to be compatible with the Apache license?

The integration parts of the algorithms must probably be rewritten to call the hipparchus numerical integration?

BTW: In my own repo I also added an implementation for hipparchus `Complex` numbers:
- https://github.com/axkr/symja_android_library/tree/master/symja_android_library/matheclipse-core/src/main/java/org/matheclipse/core/numerics/series

**Comment by maisonobe** [comment]
We could not incorporate them as is due to the license. Hipparchus is published under the Apache license and numerical-integration is published under the LGPL license.
If the original author is willing to change the license terms so we can publish as Apache V2, then sure, this kind of algorithms could be included in Hipparchus.

Concerning implementation of numerical integration, perhaps we should look if we could include the integration algorithms too.
